util/symbols.py

```diff
@@ -250,6 +252,13 @@ def get_seg_for_rom(rom: int) -> Optional["Segment"]:
                 sym.segment.add_symbol(sym)
 
             sym.user_declared = True
+            if (addr, sym.rom) in seen_symbols:
+                log.parsing_error_preamble(path, line_num, line)
+                log.error(
+                    f"Duplicate symbol detected ({sym}). 0x{addr:X} has already a symbol definition"
+                )
+
+            seen_symbols.add((addr, sym.rom))
```

Comment:
So this only checks for duplicated symbols that have both same vram and same rom, right?
It is a bit of a limited check, like it doesn't check for duplicated names

Reply:
Is it an issue to have the same symbol name with different vram+rom? Initially I was checking for name+vram+rom, but thought that was too specific.

I guess the question is: what use cases are we trying to avoid.

```
var1 = address1
var1 = address2 (guess we should check and error on this?)
var2 = address1 (presumably we don't want a user doing this?)
```

But is this invalid:
```
var1 = address1; // rom:rom1
var1 = address2 // rom:rom2 
```

Reply:
Duplicated symbol names is definitely invalid, it will produce a link time error for duplicated symbol names
```
var1 = address1;
var1 = address2;
```

Declaring the same variable more than once (even if both are valid) could be checked by splat too, since it can be error-prone, forgetting to rename the second one
```
var1 = address1;
var1 = address1;
```

Duplicated vram addresses are fine because of overlapping overlays and yada yada.

I guess we could also check for duplicated rom addresses now that we are on it
```
var1 = address1; // rom:rom_address_1
var2 = address2; // rom:rom_address_1 error!
```

Reply:
About to commit some changes.

**Check for address clash (no rom specified):**
```ini
play_sound_effect_at_location    = 0x8001BA00;
play_sound_effect_at_location2   = 0x8001BA00;
```
produces
```
error reading symbol_addrs.esa.txt, line 15:
	play_sound_effect_at_location2   = 0x8001BA00;
Duplicate symbol detected! play_sound_effect_at_location2 clashes with play_sound_effect_at_location defined at 0x8001BA00
```

**Check for name clash**
```ini
play_sound_effect_at_location    = 0x8001BA00;
play_sound_effect_at_location    = 0x8001BA00;
; or alternatively
play_sound_effect_at_location    = 0x8001BA00;
play_sound_effect_at_location    = 0x8001BA20;

```
produces
```
error reading symbol_addrs.esa.txt, line 15:
	play_sound_effect_at_location    = 0x8001BA00;
Duplicate symbol detected! play_sound_effect_at_location has already been defined at 0x8001BA00
```

**Symbols that share vram but have different rom address will pass the checks:**
```ini
play_sound_effect_at_location    = 0x8001BA00; // rom:0x1000
play_sound_effect_at_location2   = 0x8001BA00; // rom:0x2000
```

Reply:
About the last one, a symbol that shares the same vram should be able to pass the check if it either have a different vrom address or a different segment

Reply:
Trying to get the logic on this right.

I think we want these example to fail:

```ini
; ambiguous rom (therefore segment) in first symbol, vram clashing with 2nd
play_sound_effect_at_location1   = 0x8001BA00;
play_sound_effect_at_location2   = 0x8001BA00; // rom:0x12000

; ambigious segment in first symbol, vram clashing with 2nd
play_sound_effect_at_location3   = 0x8001BB00;
play_sound_effect_at_location4   = 0x8001BB00; // segment:main

```

which leads me to this logic which I want to run by you before I commit another bit of wrong code!
```py
if addr in all_symbols_dict:
    items = all_symbols_dict[addr]
    for item in items:
        if (
            sym.rom == item.rom
            or None in (sym.rom, item.rom)
            or sym.segment == item.segment
            or None in (sym.segment, item.rom)
        ):
            log.parsing_error_preamble(path, line_num, line)
            log.error(
                f"Duplicate symbol detected! {sym.name} clashes with {item.name} defined at 0x{addr:X}"
            )
```